src/com/telesign/util/TeleSignRequest.java

```diff
@@ -370,7 +370,7 @@ public String getPostBody() {
 	 *			 error occurred while attempting to sign the Request.
 	 */
 	public String executeRequest() throws IOException {
-
+		setSigningMethod(auth);

```

Comment:
Unless I am missing something this could be a problem. If the user calls setSigningMethod to set auth the method will add x-ts-auth-method but then we call it again during the executeRequest so it will add the header twice. Also I don't see any unit test to prove this doesn't break something.